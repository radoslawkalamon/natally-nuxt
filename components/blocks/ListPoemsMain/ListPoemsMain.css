.list-poems-main {
  align-items: center;
  display: flex;
  flex-flow: column nowrap;
  gap: var(--gap);
}
