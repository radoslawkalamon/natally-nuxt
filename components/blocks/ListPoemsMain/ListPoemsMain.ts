import Vue from 'vue'

export default Vue.extend({
  name: 'BlocksListPoemsMain',
  components: {
    ComponentsCardPoem: () => import(/* webpackChunkName: "components-card-poem" */'@/components/components/CardPoem/CardPoem.vue'),
    WrappersListPoems: () => import(/* webpackChunkName: "wrappers-list-poems" */'@/components/wrappers/ListPoems/ListPoems.vue')
  }
})
