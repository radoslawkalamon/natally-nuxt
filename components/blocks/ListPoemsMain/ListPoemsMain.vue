<template>
  <WrappersListPoems
    class="list-poems-main"
    data-test="blocks-list-poems-main"
  >
    <template #loader>
      <template v-for="i in 6">
        <ComponentsCardPoemPlaceholder :key="i" />
      </template>
    </template>
    <template #default="{ metaPostPoems }">
      <template v-for="metaPostPoem in metaPostPoems">
        <ComponentsCardPoem
          :key="metaPostPoem.path"
          :meta-post-poem="metaPostPoem"
        />
      </template>
    </template>
  </WrappersListPoems>
</template>

<script src="./ListPoemsMain.ts" lang="ts"></script>
<style src="./ListPoemsMain.css" scoped></style>
