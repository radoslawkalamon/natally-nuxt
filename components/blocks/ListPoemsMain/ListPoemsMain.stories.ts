import { Meta, Story } from '@storybook/vue'
import ListPoemsMain from './ListPoemsMain.vue'

const meta: Meta = {
  title: 'Blocks / List Poems Main',
  component: ListPoemsMain
}
export default meta

export const Default: Story = () => ({
  components: { ListPoemsMain },
  template: '<div class="__storybook-section__"><ListPoemsMain /></div>'
})
