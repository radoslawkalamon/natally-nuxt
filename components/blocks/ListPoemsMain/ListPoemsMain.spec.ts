import { merge } from 'lodash'
import ListPoemsMain from './ListPoemsMain.vue'
import { shallRender } from '@/devtools/jest.shared.spec'

describe('Blocks / List Poems Main', () => {
  shallRender(ListPoemsMain)
})

const defaultOptionsFactory = (options?: object) => merge({
  stubs: [
    'WrappersListPoems',
    'ComponentsCardPoem'
  ]
}, options)

describe('Blocks / List Poems Main', () => {
  shallRender(ListPoemsMain, defaultOptionsFactory())
})
