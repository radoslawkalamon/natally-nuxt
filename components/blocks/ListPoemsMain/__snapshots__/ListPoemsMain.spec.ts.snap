// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blocks / List Poems Main Integration shall pass integration sanity test 1`] = `"<div class=\\"list-poems list-poems-main\\"></div>"`;

exports[`Blocks / List Poems Main Unit shall pass unit sanity test 1`] = `"<wrapperslistpoems-stub class=\\"list-poems-main\\"></wrapperslistpoems-stub>"`;
